docs/topics/testing/advanced.txt

```diff
@@ -67,6 +67,35 @@ The following is a simple unit test using the request factory::
             response = MyView.as_view()(request)
             self.assertEqual(response.status_code, 200)
 
+.. _topics-testing-advanced-allowed-hosts:
+
+Tests and ``ALLOWED_HOSTS``
+===========================
+
+.. versionadded:: 1.11
+
+If your application includes unit tests with custom host names, those tests
```

Comment:
@charettes Do you have any thoughts on this?

I suppose we don't need to have an opinion about whether folks use ``override_settings()`` or place the setting in their own custom test settings file. Looking through the docs, I don't see us making anything more than an offhand reference to this (separate settings file for tests), although I imagine a lot of well-established projects do this.

IIRC, the reason we removed ALLOWED_HOSTS from the test settings file for Django itself was that someone (possibly @charettes ?) thought it was odd for the test settings file contents to depend on a very small subset of Django's test suite, and I agreed. The same could be said of any other project’s test suite. The goal was to keep anything that *all* of Django’s tests care about in the test settings file, and keep anything that’s specific to just a few of the tests close to (in the same file with) those tests.

I also don't see this as adding much/any boilerplate (perhaps one line to the top of the 1-2 classes in your project that test multi-domain functionality); this just requires that test writers make sure their settings are set up correctly for tests, just as they would for any other setting in Django. I think the minimal nature of changes required to Django’s own test suite to support enforcing ALLOWED_HOSTS during tests demonstrates this.